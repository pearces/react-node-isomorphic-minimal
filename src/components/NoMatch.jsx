import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import './NoMatch.scss';

const NoMatch = ({ routes = [] }) => (
  <>
    <h1>404 - Not Found</h1>
    <h3>Try one of these instead:</h3>
    <ul className="no-list-style">
      {routes.map(({ path }) => (
        <li key={path}>
          <Link to={path}>{path}</Link>
        </li>
      ))}
    </ul>
  </>
);

NoMatch.propTypes = {
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string.isRequired,
      component: PropTypes.element
    })
  )
};

export default NoMatch;
