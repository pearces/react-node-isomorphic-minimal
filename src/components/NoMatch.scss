ul.no-list-style {
  list-style-type: none;
  padding: 0;
}
