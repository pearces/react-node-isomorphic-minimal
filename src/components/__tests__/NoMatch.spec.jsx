import React from 'react';
import { MemoryRouter as Router } from 'react-router';
import { render, screen } from '@testing-library/react';
import NoMatch from '../NoMatch';

describe('NoMatch', () => {
  const routes = [
    { path: '/', element: <div /> },
    { path: '/foo', element: <div /> }
  ];

  it('matches component snapshot', () => {
    const html = render(
      <Router>
        <NoMatch routes={routes} />
      </Router>
    );
    const tree = html.asFragment();
    expect(tree).toMatchSnapshot();
  });

  it('renders routes as links', () => {
    render(
      <Router>
        <NoMatch routes={routes} />
      </Router>
    );
    expect(screen.queryAllByRole('link').length).toEqual(routes.length);
  });
});
