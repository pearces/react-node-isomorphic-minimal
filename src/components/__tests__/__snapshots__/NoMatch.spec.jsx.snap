// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NoMatch matches component snapshot 1`] = `
<DocumentFragment>
  <h1>
    404 - Not Found
  </h1>
  <h3>
    Try one of these instead:
  </h3>
  <ul
    class="no-list-style"
  >
    <li>
      <a
        data-discover="true"
        href="/"
      >
        /
      </a>
    </li>
    <li>
      <a
        data-discover="true"
        href="/foo"
      >
        /foo
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
